src/main/java/com/tabnine/inline/render/GraphicsUtils.kt

```diff
@@ -15,6 +17,43 @@ import kotlin.math.abs
 import kotlin.math.sqrt
 
 object GraphicsUtils {
+    private var tabSize: Int? = null
+
+    fun getTabSize(editor: Editor): Int? {
+        if (tabSize != null) {
+            return tabSize
+        }
+
+        // Some tests don't run with read access -> can't access tabSize information
+        if (ApplicationManager.getApplication().isUnitTestMode) {
+            return DEFAULT_TAB_SIZE
+        }
+
+        AppExecutorUtil.getAppExecutorService().submit {
+            ApplicationManager.getApplication().runReadAction {
+                Logger.getInstance("GraphicsUtils").info("Reading tab size from editor")
+                tabSize = readTabSize(editor)
```

Comment:
Hmnm.... I think tabSize can differ in IJ per language. That's why you pass in the editor.

Not sure it will work correctly to store a global tab size

Reply:
i think you're right 🙂 
i can always issue a read operation and return the previous value, wdyt about that?
also, before the bugfix it was always returning 4 and i didn't hear many complains about it - i guess 99% of users have tabSize = 4 anyways...

Reply:
Previous value might still be a different editor, so it is istill not 100% correct.

Can't we request read access? Or maybe call this function fro a different place that has read access?